cms/models.py

```diff
@@ -252,6 +259,13 @@ def clean(self):
             if errors:
                 raise ValidationError(errors)
 
+    def get_context(self, request, *args, **kwargs):
+        return {
+            **super().get_context(request),
+            **get_base_context(request),
+            "coursepage": CoursePage.objects.get(course=self.course),
```

Comment:
We could add an extra check to avoid the object not found error as there can be courses with no course detail page. Also, how about we just pass the page URL here instead of passing the course object?